maxima/g0/neutrals/sigma-cx-calc.mac

```diff
@@ -0,0 +1,141 @@
+/* Generate CX cross section (vSigmaCX) kernels for charge
+   exchange operator */
+
+load("modal-basis");
+load("out-scripts");
+load(stringproc)$
+fpprec : 24$
+
+/* Serendipity basis. */
+maxPolyOrder_Ser : 2$
+
+declareWriteCExprsC(lhs, rhs) := block([expr],
+  expr : float(expand(rhs)),
+  if expr # 0.0 then (
+    ep : string(expr),
+    for i : 0 thru 60 do (
+    	ep : ssubst(string(pow(u_ion[i],2.)),string(u_ion[i]^2),ep),
+    	ep : ssubst(string(pow(u_neut[i],2.)),string(u_neut[i]^2),ep)	
+    ),
+    printf(fh, "  double ~a = ~a; ~%", lhs, ep)
+  )
+)$
+
+writeCExprsCs(lhs, rhs) := block([expr],
+  expr : float(expand(rhs)),
+  for i : 1 thru length(expr) do (	
+    if expr[i] # 0.0 then (
+      ep : string(expr[i]),
+      ep : ssubst("log(","ln(",ep),	
+      printf(fh, "  ~a = ~a; ~%", lhs[i-1], ep)	
+    )
+  )
+)$
+
+writeCExprsC(lhs, rhs) := block([expr],
+  expr : float(expand(rhs)),
+  if expr # 0.0 then (
+     ep : string(expr),
+     printf(fh, "  ~a = ~a; ~%", lhs, ep)	
+  )
+)$
+
+/* vSigmaCX for charge exchange cross section */
+calcSigmaCXcellAv(fh, funcNm, cdim, vdim, basisFun, pMax) := block([],
+  kill(varsC, varsP, basisC, basisP),
+  modNm : sconcat("basis-precalc/basis", basisFun, cdim, "x", vdim, "v"),
+  load(modNm),
+  for p : 1 thru pMax do (
+    bP : basisP[p],
+    print(basisFun, cdim, vdim, p, length(basisP)),
+    bC : basisC[p],
+    NC : length(bC),
+
+    printf(fh, " ~%"),
+    printf(fh, "GKYL_CU_DH static inline ~%"),
+    printf(fh, "double ~a_p~a(const double a, const double b, double vt_sq_ion_min, double vt_sq_neut_min, const double *m0, const double *prim_vars_ion, const double *prim_vars_neut, double* GKYL_RESTRICT v_sigma_cx) ~%{ ~%", funcNm, p),
+    printf(fh, "  // a               constant in fitting function. ~%"),
+    printf(fh, "  // b               constant in fitting function. ~%"),
+    printf(fh, "  // m0[~a]:         neutral particle density. ~%", NC),
+    printf(fh, "  // prim_vars_ion[~a]:   ion prim vars. ~%", NC*(vdim+1)),
+    printf(fh, "  // prim_vars_neut[~a]:  neut prim vars. ~%", NC*(vdim+1)),
+    printf(fh, "  // v_sigma_cx:          cell ave cross section fitting eqn. ~%"),
+    printf(fh, " ~%"),
+
+    m0_e : doExpand1(m0,bC),
+    m0_av :  innerProd(varsC,1.0,1.0,m0_e)/innerProd(varsC,1.0,1.0,1.0),
+    declareWriteCExprsC(m0_neut_av, m0_av),
```

Comment:
at some point it'd be nice to add functions like these, that declare, to out-scripts.mac